docker/gcp_pudl_etl.sh

```diff
@@ -107,6 +111,12 @@ copy_outputs_to_gcs
 
 # if pipeline is successful, distribute + publish datasette
 if [[ $ETL_SUCCESS == 0 ]]; then
+    if [ $GITHUB_ACTION_TRIGGER = "schedule" ]; then
+        # Tag the nightly build
+        git config user.email "pudl@catalyst.coop"
+        git config user.name "PudlBot"
+        git tag -a -m "The most recent successful nightly build." nightly $GITHUB_REF
+        git push origin nightly
```

Comment:
- I don't think this will work, because I don't think what we have inside the Docker container is a git repo.
- We also need to somehow make use of the `PUDL_BOT_PAT` to authenticate here.
- Also, we want `nightly` to be a branch, not a tag, so the action we take here needs to be a merge or a rebase or something rather than just a tag.

Reply:
The dockerfile does copy over `.git/` so it might work?

Reply:
The dockerfile *mounts* `.git` explicitly to avoid copying the repo history etc. over - so I think you're right about the not-a-repo thing.

I think you're right about using the PAT in place of a password: https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/managing-your-personal-access-tokens#using-a-personal-access-token-on-the-command-line

We can use the PAT to do a fresh `git clone` and do all our fancy git stuff using that copy of the repo, I think.

Reply:
@jdangerx do you mean that **instead** of copying over the file hierarchy when building the container, we should do like `git clone --depth 2` inside the Docker container when it's running, since we're going to need to have git credentials inside it anyway?  Or that we should do the `git clone` in the Dockerfile so it happens when we're building the image? Or that we should both copy the repository files in, and then also `git clone` on top of them?

Looking at the `Dockerfile`:

```Dockerfile
# Copy dagster configuration file
COPY docker/dagster.yaml ${DAGSTER_HOME}/dagster.yaml

# Create a conda environment based on the specification in the repo
COPY environments/conda-lock.yml environments/conda-lock.yml
RUN micromamba create --prefix ${CONDA_PREFIX} --yes --file environments/conda-lock.yml && \
    micromamba clean -afy
# Copy the cloned pudl repository into the user's home directory
COPY --chown=${MAMBA_USER}:${MAMBA_USER} . ${CONTAINER_HOME}
```

- It seems like we need to copy the `dagster.yaml` into the appropriate place, since it's not inside the repo. But it could be copied to the correct location from within the container rather than from a cloned repo outside the container. Is there still a need to check out the repo in the github workflow? Or could it only happen inside the Docker container?
- If we're copying the entire repository into the image, why is `conda-lock.yml` called out specifically?
- Does the final `COPY` of `.` **not** get the `.git` directory? If it **does** get the `.git` directory, then why are we subsequently mounting it before installing PUDL:

```Dockerfile
# We need information from .git to get version with setuptools_scm so we mount that
# directory without copying it into the image.
RUN --mount=type=bind,source=.git,target=${PUDL_REPO}/.git \
    ${CONDA_RUN} pip install --no-cache-dir --no-deps --editable .
```

If we need the whole repository inside the Docker container, could we do something like this in the Dockerfile?

```Dockerfile
RUN git clone --depth 2 --single-branch --branch $GITHUB_REF https://github.com/catalyst-cooperative/pudl.git
RUN git remote set-url origin https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git
RUN git config user.email "pudl@catalyst.coop"
RUN git config user.name "pudlbot"
```